website/source/docs/agent/caching/index.html.md

```diff
@@ -13,55 +13,77 @@ description: |-
 token, then the `auto-auth` token will used to make requests. The resulting
-secrets from these `auto-auth` token calls are not cached. They will be in the
-non-beta version. To test out the caching scenarios, please make a login
-request or a token creation request via the agent. The secrets generated from
-these new tokens will get cached.
+secrets from these `auto-auth` token calls however are not cached. They will be
+cached in the non-beta version. To test the caching scenarios, please make
+a login request or a token creation request via the agent. The secrets
+generated from these new tokens will get cached.
 
 ## Cache Evictions
 
-The eviction of cache entries will occur when the agent fails to renew secrets.
-This can happen when the secret that is cached hits it's maximum TTL or if the
-renewal results in an error.
+The eviction of cache entries pertaining to any secret will occur when the
+agent can no longer renew it.  This can happen when the cached secret hits it's
+maximum TTL or if the renewal results in any error.
 
 Agent also does some best-effort cache evictions by observing specific request
 types and response codes. For example, if a token revocation request is made
 via the agent and if the request succeeds, then agent evicts all the cache
-entries associated with the revoked token. Similar behavior is exercised for
-lease revocations as well.
-
-While agent tries to observe some requests and evicts cache entries
-automatically, agent is completely unaware of revocations that happen outside of
-Agent's context. This is when stale entries are created in the agent.
-
-For managing the stale entries in the cache, an endpoint
-`/v1/agent/cache-clear`(see below) is made available to manually evict cache
-entries based on some of the criteria.
+entries associated with the revoked token. Similarly, any lease revocation
+operation will be intercepted by the agent and respective cache entries will be
+evicted.
+
+Note that while agent evicts the cache entries when the secret expires
+and by observing some revocation request pattern, it will be completely unaware
+of the revocations that happen through direct interactions with the Vault
+server. This could potentially lead to stale cache entries. For managing the
+stale entries in the cache, an endpoint `/v1/agent/cache-clear`(see below) is
+made available to manually evict cache entries based on some of the criteria.
+
+## Request Uniqueness
+
+In order to detect a repeat request and return cached response, agent will need
+to have a way to uniquely identify the requests. This computation as it stands
+now takes a simplistic approach (may change in future). The HTTP request that
+is received by the agent is serialized and hashed. This hash value is used as
+an index into the cache to check if the response is readily available. This
+implies that the hash value will differ if any data in the request is modified.
```

Comment:
Maybe something like this to make it more explicit, and that ordering is one case where this can happen, though not the only one. 

> This implies that the hash value will differ if any data in the request is modified, such as the ordering of the request body. This, in turn, will result in agent caching the request as a distinct entry. As long as requests come in without any change, caching behavior should be consistent.

Reply:
See if the update I pushed addresses this.